relay-server/src/actors/upstream.rs

```diff
@@ -881,69 +890,115 @@ impl Entry {
 /// is synchronous and managed by the [`UpstreamBroker`].
 #[derive(Debug)]
 struct UpstreamQueue {
+    /// High priority queue.
     high: VecDeque<Entry>,
+    /// Low priority queue.
     low: VecDeque<Entry>,
+    /// Retry queue, sorted by [priority][`RequestPriority`].
+    retries: VecDeque<Entry>,
```

Comment:
Instead of a single retry queue, this needs to have one for high and one for low priority. Otherwise, you break queue order on retries, since you cannot simply push_front into the retry queue.